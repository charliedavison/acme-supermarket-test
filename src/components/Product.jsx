import React from 'react';
import PropTypes from 'prop-types';

class Product extends React.Component {
    constructor(props) {
        super(props);
        this.onClick = this.onClick.bind(this);
    };

    onClick() {
        this.props.onClick(this.props.product.productCode);
    }

    render() {
        return (
            <div onClick={this.onClick} className="product">
                <img src={`img/${this.props.product.image}`} className="product-image" />
                <div className="product-name">{this.props.product.name}</div>
            </div>
        )
    }
};
Product.propTypes = {
    product: PropTypes.shape({
        productCode: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired,
        price: PropTypes.number.isRequired,
        image: PropTypes.string.isRequired
    }).isRequired,
    onClick: PropTypes.func.isRequired
}
module.exports = Product;