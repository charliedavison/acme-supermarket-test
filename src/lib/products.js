module.exports = {
    'FR1': {
        productCode: 'FR1',
        name: 'Fruit tea',
        price: 3.11,
        image: 'tea.jpg'
    },
    'SR1': {
        productCode: 'SR1',
        name: 'Strawberries',
        price: 5.00,
        image: 'strawberries.jpg'
    },
    'CF1': {
        productCode: 'CF1',
        name: 'Coffee',
        price: 11.23,
        image: 'coffee.jpg'
    }
};