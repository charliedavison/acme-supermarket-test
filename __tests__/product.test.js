import React from 'react';
import { shallow } from 'enzyme';
import Product from './../src/components/Product.jsx';
import products from './../src/lib/products.js';

describe('<Product />', () => {
    let component;

    let props = {
        product: products.FR1,
        onClick: jest.fn()
    };

    beforeEach(() => {
        component = shallow(<Product {...props} />);
    });

    test('Should match snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    test('Clicking component should call the passed in callback', () => {
        expect(props.onClick).toHaveBeenCalledTimes(0);
        component.find('.product').simulate('click');
        expect(props.onClick).toHaveBeenCalledTimes(1);
    });
});