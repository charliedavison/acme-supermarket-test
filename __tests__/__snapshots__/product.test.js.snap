// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Product /> Should match snapshot 1`] = `
<div
  className="product"
  onClick={[Function]}
>
  <img
    className="product-image"
    src="img/tea.jpg"
  />
  <div
    className="product-name"
  >
    Fruit tea
  </div>
</div>
`;
